### ISSUE TITLE
 [tune] quniform sampling results out of bound

### ISSUE THREAD

**Issue by meechos** [ray-project/ray#12717]
### What is the problem?
 The quantised uniform sampling results out of bound.  A dropout range of 

```
dropout = tune.quniform(0.0, 0.9, 0.1)
>1.08
```

*Ray version and other system information (Python version, TensorFlow version, OS):*
ray 1.0.1-post1
torch 1.4.0
python 3

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by ericl** [comment]
cc @richardliaw

**Comment by krfricke** [comment]
Hi @meechos, can you give a reproduction script? I can't reproduce the issue:

```
from ray import tune

dropout = tune.quniform(0.0, 0.9, 0.1)
sample = dropout.sample(size=1000000)
print(f"Min: {min(sample)} Max: {max(sample)}")
# Result: Min: 0.0 Max: 0.9
```

**Comment by krfricke** [comment]
Are you using PopulationBasedTraining by any chance? If so, please paste how you instantiate the PBT scheduler and how you call tune.run.

In short, PBT does hyperparameter mutations by multiplying existing parameters value by 0.8 or 1.2 - hence 1.08 (which is 0.9 * 1.2). This is expected behavior as discussed in the original PBT paper.

**Comment by krfricke** [comment]
Closing because of inactivity - please feel free to reopen if this issue is not related to PBT or if you think we should introduce more thorough checks for this in PBT (in that case, please start a discussion on https://discuss.ray.io/ so others are encouraged to join in the discussion).
